### PR TITLE
[INLONG-7816][Manager] Support validation rules when exporting Excel file

### LINKED ISSUE #7816: [Feature][Manager] Support validation rules when exporting Excel file
### Description

image

When generating templates, it is necessary to restrict validation rules such as field types and enumeration values to avoid user input errors.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by featzhang**
Fixes #7816 

## Validation results

image

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 1]
```diff
@@ -159,6 +168,54 @@ private static void fillSheetHeader(XSSFRow row, List<String> heads) {
...
+     * @param sheet  the sheet to fill with validation constraints
+     * @param fields the fields to use for validation constraints
+     */
+    private static <T> void fillSheetValidation(Class<T> clazz, XSSFSheet sheet, Field[] fields) {
+        List<Pair<String, ExcelField>> excelFields = Arrays.stream(fields).map(field -> {
+            field.setAccessible(true);
+            return Pair.of(field.getName(), field.getAnnotation(ExcelField.class));
+        }).filter(p -> p.getRight() != null).collect(Collectors.toList());
+
+        IntStream.range(0, excelFields.size())
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:184]: Is a for-loop enough?

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]
LGTM

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 2]
```diff
@@ -159,6 +168,59 @@ private static void fillSheetHeader(XSSFRow row, List<String> heads) {
...
+                    .filter(v -> v != ExcelCellValidator.class)
+                    .map(v -> {
+                        try {
+                            return (ExcelCellValidator<?>) v.newInstance();
+                        } catch (InstantiationException | IllegalAccessException e) {
+                            LOGGER.error("Can not properly create ExcelCellValidator", e);
+                            return null;
+                        }
+                    })
+                    .map(ExcelCellValidator::constraint);
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:204]: Why call map twice？

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 3]
```diff
@@ -159,6 +168,59 @@ private static void fillSheetHeader(XSSFRow row, List<String> heads) {
...
+            field.setAccessible(true);
+            ExcelField excelField = field.getAnnotation(ExcelField.class);
+            if (excelField != null) {
+                excelFields.add(Pair.of(field.getName(), excelField));
+            }
+        }
+
+        int bound = excelFields.size();
+        for (int index = 0; index < bound; index++) {
+            Pair<String, ExcelField> se = excelFields.get(index);
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:191]: Why is it named `se`？

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 4]
```diff
@@ -159,6 +168,59 @@ private static void fillSheetHeader(XSSFRow row, List<String> heads) {
...
+            }
+        }
+
+        int bound = excelFields.size();
+        for (int index = 0; index < bound; index++) {
+            Pair<String, ExcelField> excelFieldPair = excelFields.get(index);
+            Class<? extends ExcelCellValidator> validator = excelFieldPair.getRight().validator();
+
+            Optional<List<String>> optionalList = Optional.ofNullable(validator)
+                    .filter(v -> v != ExcelCellValidator.class)
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:195]: Is this statement valid？

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java [hunk 5]
```diff
@@ -159,6 +168,59 @@ private static void fillSheetHeader(XSSFRow row, List<String> heads) {
...
+                excelFields.add(Pair.of(field.getName(), excelField));
+            }
+        }
+
+        int bound = excelFields.size();
+        for (int index = 0; index < bound; index++) {
+            Pair<String, ExcelField> excelFieldPair = excelFields.get(index);
+            Class<? extends ExcelCellValidator> validator = excelFieldPair.getRight().validator();
+
+            Optional<List<String>> optionalList = Optional.ofNullable(validator)
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/tool/excel/ExcelTool.java:194]: `Optional.ofNullable(validator)` maybe is null.

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**